web-common/src/features/dashboards/selectors/index.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  createQueryServiceMetricsViewToplist,
+  createRuntimeServiceGetCatalogEntry,
+  type V1MetricsView,
+} from "@rilldata/web-common/runtime-client";
+import type { BusinessModel } from "../business-model/business-model";
+import { derived, writable } from "svelte/store";
+import type { CreateQueryResult } from "@tanstack/svelte-query";
+
+export const useMetaQuery = <T = V1MetricsView>(
+  ctx: BusinessModel,
+  selector?: (meta: V1MetricsView) => T
+) => {
+  return derived(
+    [ctx.runtime, ctx.metricsViewName],
+    ([runtime, metricViewName], set) => {
+      return createRuntimeServiceGetCatalogEntry(
+        runtime.instanceId,
+        metricViewName,
+        {
+          query: {
+            select: (data) =>
+              selector
+                ? selector(data?.entry?.metricsView)
+                : data?.entry?.metricsView,
+            queryClient: ctx.queryClient,
+          },
+        }
+      ).subscribe(set);
+    }
+  );
+};
```

Comment:
Should we use the [v5 reactivity pattern](https://tanstack.com/query/v5/docs/svelte/reactivity) and put the derived store _within_ the `createQuery` (or `createOrvalGeneratedQuery`) call?

Reply:
Really interesting idea that I hadn't considered. Let's discuss as a team, I see pros and cons:
- pro: will make moving to sq5 easier later
- cons: will require more code changes as we refactor here. Not clear yet how exactly we want to use sq5 API so we may be guessing at this point

Reply:
Would we need to make tweaks to orval generator? Does it even allow us to do this?